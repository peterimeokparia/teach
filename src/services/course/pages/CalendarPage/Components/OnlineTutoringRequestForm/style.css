.OnlineHelp h1 {
  font-size: 28px;
  text-align: center;
}
.OnlineHelp form {
  max-width: 600px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  padding: 0 20px;
  margin: 0 auto;
}
.OnlineHelp form input {
  width: 100%;
  border: 1px solid #c244d8;
  outline: none;
  transition: box-shadow 0.2s;
  margin-bottom: 20px;
}
.OnlineHelp form input:focus {
  box-shadow: 0 0 3px #c244d8;
}
.OnlineHelp form label {
  margin-bottom: 0.6em;
  display: block;
}
.OnlineHelp form button {
  color: #fff;
  background-color: #c244d8;
  font-size: 16px;
  padding: 10px;
  border-radius: 2px;
  width: 200px;
  margin: 0 auto;
  display: block;
  border: none;
  border-bottom: 3px solid #673ab7;
  transition: all 0.2s;
}
.OnlineHelp form button:focus {
  outline: none;
  box-shadow: 0 0 4px #673ab7;
}
.OnlineHelp form button:active {
  border-bottom: none;
  margin-top: 2px;
  box-shadow: 0 2px 0 #673ab7 inset;
}
.OnlineHelp form button:disabled {
  background-color: #dec5e2;
  border-bottom: 2px solid #f0e9fd;
}
.OnlineHelp .center {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}
.OnlineHelp .center .hide-button-options {
  visibility: hidden;
  opacity: 0;
  transition: visibility 0s 2s, opacity 2s linear;
}
.OnlineHelp .center .show-quick-intro-form {
  visibility: visible;
  opacity: 1;  
  transition: visibility 0s 2s, opacity 2s linear;
  /* transition: opacity 2s linear; */
}
.OnlineHelp .center .show-quick-intro-form-hidden {
  visibility: hidden;
  /* opacity: 0;
  transition: visibility 0s 2s, opacity 2s linear; */
}
.OnlineHelp .show-quick-intro-form-test {
  visibility: visible;
  opacity: 1;  
  transition: visibility 0s 2s, opacity 2s linear;
  /* transition: opacity 2s linear; */
}
.OnlineHelp .center .blink {
  width: 520px;
  /* height: 50px; */
  /* background-color: #342ab8; */
  padding: 10px;
  padding-bottom: 10px;
  text-align: center;
  line-height: 50px;
}
.OnlineHelp .center .blink div {
  font-size: 26px;
  font-family: cursive;
  color: black;
  /* color: #fff; */
  animation: blink 2s linear infinite;
}
@keyframes blink {
  0% {
    opacity: 0;
  }
  50% {
    opacity: .5;
  }
  100% {
    opacity: 1;
  }
}