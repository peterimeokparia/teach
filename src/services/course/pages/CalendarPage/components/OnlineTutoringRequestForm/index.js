import { 
connect } from 'react-redux';

import { 
getOperatorFromOperatorBusinessName, 
getUsersByOperatorId,
getCoursesByOperatorId } from 'services/course/selectors';

import useOnlineTutoringRequestFormHook from 'services/course/pages/CalendarPage/hooks/useOnlineTutoringRequestFormHook';
import DropDown from 'services/course/pages/components/DropDown';
import './style.css';

const OnlineTutoringRequestForm = ({
    saveInProgress,
    onSaveError,
    calendarEventType,
    user,
    tutors,
    courses,
    operatorBusinessName,
    operator,
    dispatch }) => {
    let{
        courseName, 
        setCourseName,
        immediateHelp, 
        setImmediateHelp, 
        getTutor, 
        setGetATutor,
        inputRef
    } = useOnlineTutoringRequestFormHook();
        
    if ( saveInProgress ) {
        return <div>...loading</div>;
    } 

    if ( onSaveError ) {
        return <div> { onSaveError.message } </div> ;
    }      

const handleSubmit = e => { 
    e.preventDefault(); 
};

const getImmediateHelp = e => { 
    setImmediateHelp( true );
    // dispatch(addNewCourse(courseName, coursePrice, courseDescription, currentUser, operator));
};

const getScheduledHelp = e => { 
    // dispatch(addNewCourse(courseName, coursePrice, courseDescription, currentUser, operator));
};

const getATutor = e => {
    setGetATutor( true );
};

return (
    <div className="OnlineHelp">
        <h1>{`.`}</h1> 
        <br></br>
        <form onSubmit={handleSubmit}>    
            <div className="center">
                <div className={( immediateHelp ) ? "hide-button-options" : ""}>
                <div className="row justify-content-sm-center">
                    <span className="col">
                        <button className="" onClick={ getImmediateHelp } disabled={ saveInProgress } > {`Immediate Assistance`} </button> 
                    </span>
                    <span className="col">
                        <button className="" onClick={ getScheduledHelp } disabled={ saveInProgress } > {`Scheduled Session`} </button> 
                    </span>
                </div>
                </div>
                {
                <div>   
                <div className={(getTutor) ? "hide-button-options" : ""}>
                {  
                    <div className={( immediateHelp ) ? "show-quick-intro-form" : "show-quick-intro-form-hidden"}>
                    <label>
                        <input
                            ref={inputRef} 
                            disabled={saveInProgress} 
                            value={courseName} 
                            onChange={(e) => setCourseName(e.target.value)}
                            placeholder="Email Address"
                        />
                    </label>
                    <label>
                        <input
                            ref={inputRef} 
                            disabled={saveInProgress} 
                            value={courseName} 
                            onChange={(e) => setCourseName(e.target.value)}
                            placeholder="First Name"
                        />
                    </label>
                    <label>
                        <div>   {'How can we assist you?'} </div>
                        <div className="row justify-content-sm-center">
                            <DropDown optionCollection={ [] }/>
                        </div>    
                    </label>
                    <button className={( immediateHelp ) ? "show-quick-intro-form" : "show-quick-intro-form-hidden"} onClick={ getATutor } disabled={ saveInProgress } > {`Get a Tutor`} </button> 
                </div>
                }
                </div>
                <div className={( getTutor ) ? "show-quick-intro-form-test" : "show-quick-intro-form-hidden"}>
                    <div className={'blink'}>  <div>  {'We are getting you connected with a tutor. Please wait...'}  </div>  </div>
                                    {/* https://www.tawk.to/ */}
                    <div> { ' Ad & Promo Rotation. Ad & Promo Rotation. Ad & Promo Rotation. Ad & Promo Rotation.  '}  </div>
                    <div> { 'While you wait, do you know that you can get a discount by signing up today? '}  </div>
                </div>
                </div>
            }
            </div>
        </form>
    </div>
); };

const mapState = ( state, ownProps ) => ({
    user: state?.users?.user, // get user by operator id
    operator: getOperatorFromOperatorBusinessName(state, ownProps),
    calendarEvents: state?.calendar?.calendarEvents, // get calendar events by operator id
    tutors: getUsersByOperatorId(state, ownProps)?.filter(user => user?.role === "Tutor"),
    courses: getCoursesByOperatorId(state, ownProps),
    saveInProgress: state.courses.saveInProgress,
    onSaveError: state.courses.onSaveError,
});

export default connect(mapState)(OnlineTutoringRequestForm);