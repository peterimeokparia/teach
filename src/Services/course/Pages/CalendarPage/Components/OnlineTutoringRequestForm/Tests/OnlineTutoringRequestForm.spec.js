import React from 'react';
import OnlineTutoringRequestForm  from 'services/course/pages/CalendarPage/components/OnlineTutoringRequestForm';
import renderer from 'react-test-renderer';
import thunk from "redux-thunk"; 
import * as reactRedux from 'react-redux';
import { Provider, useSelector } from "react-redux";
import { applyMiddleware } from 'redux'; 
import configureStore from 'redux-mock-store';

// jest.mock('../../../Api');
const operatorBusinessName = "teach";
const mockStore = configureStore([]);

describe('OnlineTutoringRequestForm', () =>  {  

 let  store = mockStore({
      saveInProgress: { courses: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      onSaveError: { courses: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} }, 
      tutors: { users: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      user: { users: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      operators: { operators: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      operator: { operators: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} },
      calendarEvents: { calendar: {_id: "5fcb0e19fd5e0117dc09dcfa", operatorBusinessName} }, 
      notifications: { pushNotificationSubscribers : {_id:"5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" } },
      users: { users: {_id: "PERSON5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      courses: { courses: {_id: "COURSE5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      grades: { grades: {_id: "COURSE5fcb0e19fd5e0117dc09dcfa", operatorBusinessName, operatorId: "5fcb0e19fd5e0117dc09dcfa" }},
      classrooms: { classrooms: {displaySideBarDropDown: false }},
      sessions: { sessions: {_id: "SESSIONS5fcb0e19fd5e0117dc09dcfa", numberOfSessions: 9, totalNumberOfSessions: 1, status: true } },
      lessons: { lessons: { selectedLessonFromLessonPlanDropDown: "Test Lesson" } },
      courses: { courses: { selectedCourseFromLessonPlanCourseDropDown: "Test Course" }  },
      meetings: { meetings: {_id: "MEETING5fcb0e19fd5e0117dc09dcfa", numberOfSessions: 9, totalNumberOfSessions: 1, status: true }  },
  }, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
      .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {

    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch)

    const tree = renderer.create(
      <Provider store={store}>
        <OnlineTutoringRequestForm 
      /> 
    </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});






