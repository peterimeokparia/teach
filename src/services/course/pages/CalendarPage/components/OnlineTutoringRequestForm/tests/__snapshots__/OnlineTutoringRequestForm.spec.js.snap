// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineTutoringRequestForm renders as expected 1`] = `
<div
  className="OnlineHelp"
>
  <h1>
    .
  </h1>
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="center"
    >
      <div
        className=""
      >
        <div
          className="row justify-content-sm-center"
        >
          <span
            className="col"
          >
            <button
              className=""
              onClick={[Function]}
            >
               
              Immediate Assistance
               
            </button>
          </span>
          <span
            className="col"
          >
            <button
              className=""
              onClick={[Function]}
            >
               
              Scheduled Session
               
            </button>
          </span>
        </div>
      </div>
      <div>
        <div
          className=""
        >
          <div
            className="show-quick-intro-form-hidden"
          >
            <label>
              <input
                onChange={[Function]}
                placeholder="Email Address"
                value=""
              />
            </label>
            <label>
              <input
                onChange={[Function]}
                placeholder="First Name"
                value=""
              />
            </label>
            <label>
              <div>
                   
                How can we assist you?
                 
              </div>
              <div
                className="row justify-content-sm-center"
              >
                <span
                  className="dropDownSelector"
                >
                  <span>
                    <form
                      onSubmit={[Function]}
                    >
                      <label>
                        <span
                          className="main-title"
                        />
                      </label>
                      <label>
                        <div
                          className="MuiInputBase-root MuiInput-root MuiInput-underline space-between-select"
                          onClick={[Function]}
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-label="Without label"
                            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onMouseDown={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <span
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "&#8203;",
                                }
                              }
                            />
                          </div>
                          <input
                            aria-hidden={true}
                            className="MuiSelect-nativeInput"
                            onAnimationStart={[Function]}
                            onChange={[Function]}
                            tabIndex={-1}
                          />
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </div>
                      </label>
                    </form>
                  </span>
                </span>
              </div>
            </label>
            <button
              className="show-quick-intro-form-hidden"
              onClick={[Function]}
            >
               
              Get a Tutor
               
            </button>
          </div>
        </div>
        <div
          className="show-quick-intro-form-hidden"
        >
          <div
            className="blink"
          >
              
            <div>
                
              We are getting you connected with a tutor. Please wait...
                
            </div>
              
          </div>
          <div>
             
             Ad & Promo Rotation. Ad & Promo Rotation. Ad & Promo Rotation. Ad & Promo Rotation.  
              
          </div>
          <div>
             
            While you wait, do you know that you can get a discount by signing up today? 
              
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
